packages/components/src/components/sidenavigation/sidenavigation.component.ts

```diff
@@ -132,8 +133,87 @@ class SideNavigation extends Provider<SideNavigationContext> {
   override connectedCallback(): void {
     super.connectedCallback();
     this.role = ROLE.NAVIGATION;
+    this.setupFlexibleOnHoverListeners();
   }
 
+  override disconnectedCallback(): void {
+    super.disconnectedCallback();
+    this.removeFlexibleOnHoverListeners();
+  }
+
+  private isHovered: boolean = false;
+
+  private isFocused: boolean = false;
+
+  /** @internal */
+  private handleMouseEnter() {
+    this.isHovered = true;
+    this.showGrabberButton();
+  }
+
+  /** @internal */
+  private handleMouseLeave() {
+    this.isHovered = false;
+    if (!this.isFocused) {
+      this.hideGrabberButton();
+    }
+  }
+
+  /** @internal */
+  private handleFocusIn(e: FocusEvent): void {
+    // Only set focus if the sidenavigation itself or a focus-visible element inside it is focused
+    if (e.target === this || (e.target as HTMLElement)?.matches(':focus-visible')) {
```

Comment:
the comment is not matching with the code. The comment says the focus-visible element is inside the sidenavigation element:


```suggestion
    if (e.target === this || (this.contains(e.target) && e.target as HTMLElement)?.matches(':focus-visible')) {
```